### ISSUE TITLE
Changes to grep inverted the condition we had set

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#12249]
@shiltemann had some published workflows (https://github.com/ErasmusMC-Bioinformatics/AMR-Galaxy-workflows) and recently they started breaking. We traced this down to the Grep1 tool, and this commit https://github.com/galaxyproject/galaxy/commit/85f643bc6630731ba810ac0b82c47a0e5ac89553

Our invert was true, it should have filtered OUT lines, instead it kept only that because it now defaulted to something else. There was a second subsequent grep step that selected for something not appearing in the new file, and we only noticed the breakage because a downstream tool rejected empty files.

On import we saw

**Step 24: Select**
- parameter 'invert': an invalid option ('true') was selected (valid options: ,-v) Using default: ''.
- No value found for 'Keep header line'. Using default: 'False'.
- Using version '1.0.3' instead of version '1.0.1' specified in this workflow. 

But it is completely unclear which of those maps to the values we expect. What does `-v` mean for someone who doesn't know the underlying grep tool parameters? What was 'true' mapped to? And since it's not possible to see the old version of this tool, you cannot even compare against an old version.

We had already handed this workflow to a clinician who isn't a galaxy expert and was running it and suddenly started having issues with it. They're competent in Galaxy but not a workflow builder and they didn't know what to make of the yellow warnings at the top and ignored it.

![image](https://user-images.githubusercontent.com/458683/125298156-3017df80-e328-11eb-84d2-8670e45e5c5f.png)

And you're not required to confirm the options in the tools which changed (they're highlighted in red, very useful!). You can run it immediately like our clinician did.

**tl;dr**
- Yes we missed messages but without really advanced galaxy knowledge it's not possible to fix those. Most people (even very competent galaxy users) will ignore those and think it's fine.
- I think we really need to keep all old versions of built in tools whenever possible

**Comment by mvdbeek** [comment]
> * I think we really need to keep all old versions of built in tools whenever possible

Yes, I think that (and more real-life workflow tests during the release testing) are the takeaway here. I will restore 1.0.1.

**Comment by bernt-matthias** [comment]
Fixed by https://github.com/galaxyproject/galaxy/pull/12252? Or do we want to document somewhere that we want to keep old tool versions.. Or manage old versions somehow else?
